lib/BitcastUtils.cpp

```diff
@@ -945,13 +961,24 @@ Value *CreateRem(IRBuilder<> &Builder, unsigned rem, Value *Val) {
   }
 }
 
+bool IsArrayLike(StructType *Ty) {
+  Type *ElemTy = Ty->getStructElementType(0);
```

Comment:
You should test for an empty struct first.

Reply:
Done